### PR TITLE
fixed virtual network subnet module not to ask for optional subnet-name… `avm/res/network/virtual-network/`

### PR DESCRIPTION
**Pull request by blinq-erik**
… input in run

## Description

Not reported as an issues
This fixes the subnet module asking for name input on run, if you run the subnet as a seperate module.

## Type of Change

added default string for optional parameters. 

## Checklist

No other issues have been reported on this

### CODE REVIEW

**Review by AlexanderSehr — COMMENTED** [review]
Hey @blinq-erik,

thats interesting - what happens if you deploy a subnet with an empty name? I did not expect for it to work.

And if it doesn't, the parameters description is wrong (as it would then be required). So something has to be fixed in either case 💪

File: avm/res/network/virtual-network/subnet/main.bicep [hunk 1]
```diff
@@ -3,7 +3,7 @@ metadata description = 'This module deploys a Virtual Network Subnet.'
 metadata owner = 'Azure/module-maintainers'
 
 @description('Optional. The Name of the subnet resource.')
```
Comment (elbatane) [inline avm/res/network/virtual-network/subnet/main.bicep:5]:
```suggestion
@description('Required. The Name of the subnet resource.')
```

File: avm/res/network/virtual-network/subnet/main.bicep [hunk 2]
```diff
@@ -3,7 +3,7 @@ metadata description = 'This module deploys a Virtual Network Subnet.'
...
-param name string
+param name string = ''
```
Comment (elbatane) [inline avm/res/network/virtual-network/subnet/main.bicep:6]:
```suggestion
param name string
```

File: avm/res/network/virtual-network/subnet/main.bicep (hunk 1 above)
Comment (elbatane) [reply avm/res/network/virtual-network/subnet/main.bicep:5]: The name is a required parameter.

File: avm/res/network/virtual-network/subnet/main.bicep (hunk 2 above)
Comment (elbatane) [reply avm/res/network/virtual-network/subnet/main.bicep:6]: @blinq-erik please review the changes, and update the title of the PR please. Thanks!

**Comment by ChrisSidebotham** [comment]
This is related to #1027 ?

**Review by eriqua — COMMENTED** [review]
@blinq-erik Are you okay to finalize this PR? Please let us know if any blocker/support needed.

**Comment by ChrisSidebotham** [comment]
reverting to draft until a response is received from @blinq-erik

**Comment by AlexanderSehr** [comment]
Given that it has been so long now and that we'll update the parameter in question in this PR anyways #3105, I'll go ahead and close this PR. @blinq-erik please feel free to re-open the PR if you feel like the issues is not addressed (after the merge)

**Pull request closed without merging**